### PR TITLE
executor: migrate test-infra to testify for merge_join_test

### LINKED ISSUE #28597: migrate test-infra to testify for `merge_join_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28597

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/merge_join_test.go [hunk 1]
```diff
@@ -221,17 +222,17 @@ const plan3 = `[[TableScan_12 {
...
 	resultStr := fmt.Sprintf("%v", result.Rows())
 	if !strings.ContainsAny(resultStr, "MergeJoin") {
-		c.Error("Expected MergeJoin in plan.")
+		t.Error("Expected MergeJoin in plan.")
 	}
 	return tk.MustQuery(sql)
 }
 
-func checkPlanAndRun(tk *testkit.TestKit, c *C, plan string, sql string) *testkit.Result {
+func checkPlanAndRun(tk *testkit.TestKit, t *testing.T, plan string, sql string) *testkit.Result {
```
Comment (mjonss) [inline executor/merge_join_test.go:235]: Please either remove the `t *testing.T` argument or fix the commented out code that currently uses `c *C`.

**Review by mjonss — APPROVED** [review]
LGTM with a minor suggestion.

File: executor/merge_join_test.go (hunk 1 above)
Comment (tisonkun) [reply executor/merge_join_test.go:235]: I'm sending a patch to address this.

**Review by tisonkun — APPROVED** [review]

**Pull request merged**